druid/src/widget/widget_ext.rs

```diff
@@ -175,6 +182,25 @@ impl<T: Data> SizedBox<T> {
     }
 }
 
+// if two things are modifying an env one after another, just combine the modifications
+impl<T: Data, W> EnvScope<T, W> {
+    pub fn env_scope(self, f2: impl Fn(&mut Env, &T) + 'static) -> EnvScope<T, W> {
+        let EnvScope { f, child } = self;
```

Comment:
for some reason I've never used this style of named struct destructuring outside of a match statement, seems handy!